packages/scoped-elements/README.md

```diff
@@ -0,0 +1,276 @@
+4. Team Green (on Page B) has to deliver an important functionality to your end users but they need to upgrade to Feature B 2.x as it can only be solved with this new version.
+5. As Feature A 1x & 2x are both used in the same app it will get installed nested which will lead to [catastrophic failure]https://open-wc.org/scoped-elements./demo/no-scoping/) [[code](https://github.com/open-wc/open-wc/tree/master/packages/scoped-elements/demo/no-scope)]. Two possible solutions come to mind:
+   1. Synchronizing updates of shared dependencies - e.g. make sure Team Blue & Green always us the same version when releasing. This can be a viable solution however it comes with a high organizational overhead and is hard to scale up (for 10+ teams)
+   2. Temporarily (!) allow to ship similar source code (most breaking releases are not a total rewrite) and scope them via `@open-wc/scoped-elements` => see "fixed" [with-scope](https://open-wc.org/scoped-elements/demo/with-scoping/) [[code](https://github.com/open-wc/open-wc/tree/master/packages/scoped-elements/demo/with-scope)] running with nested dependencies.
+
+#### Technical explanation of scenario
+
+The simplified app has the following dependencies
+
+- app
+  - page-a
+    - feature-a 1.x
+    - feature-b 1.x
+  - page-b
+    - feature-a 2.x
+    - feature-b 1.x
+
+which leads to the following node_modules tree
+
+```
+├── node_modules
+│   ├── feature-a
+│   ├── feature-b
+│   ├── page-a
+│   └── page-b
+│       └── node_modules
+│           └── feature-a
+├── demo-app.js
+└── index.html
+```
+
+So there are 3 demos:
+
+1. [before-nesting](https://open-wc.org/scoped-elements/demo/before-nesting/) [[code](https://github.com/open-wc/open-wc/tree/master/packages/scoped-elements/demo/before-nesting)] everything fine as Page A/B both are using the same version of Feature A
+
+2. [no-scope](https://open-wc.org/scoped-elements/demo/no-scoping/) [[code](https://github.com/open-wc/open-wc/tree/master/packages/scoped-elements/demo/no-scope)] Feature A version 1.x and 2.x are imported via self registering entry points which leads to the following error message as "feature-a" tries to register multiple times
+
+   ```
+   Uncaught DOMException: Failed to execute 'define' on 'CustomElementRegistry': the name "feature-a" has already been used with this registry
+     at [...]/node_modules/page-b/node_modules/feature-a/feature-a.js:3:16
+   ```
+
+3. [with-scope](https://open-wc.org/scoped-elements/demo/with-scoping/) [[code](https://github.com/open-wc/open-wc/tree/master/packages/scoped-elements/demo/with-scope)] Successfully "fixing" the demo by using `createScopedHtml` on Page A/B.
+
+## How it works
+
+`createScopedHtml` is going to define the custom element classes and creates a
+unique scoped tag for each of them if needed. In case those classes were
+previously defined then it's going to return the previous defined tag.
+
+Then, the `html` function provided by the `createScopedHtml` method is going to
```

Comment:
This really needs a better explanation. Can someone take a look at this? @manolakis @daKmoR @LarsDenBakker

Reply:
I applied @thepassle suggestion with a better explanation.